llama-index-core/llama_index/core/base/llms/types.py

```diff
@@ -116,8 +149,16 @@ def _recursive_serialization(self, value: Any) -> Any:
     def serialize_additional_kwargs(self, value: Any, _info: Any) -> Any:
         return self._recursive_serialization(value)
 
-    def dict(self, **kwargs: Any) -> Dict[str, Any]:
-        return self.model_dump(**kwargs)
+    # @field_serializer("blocks", check_fields=False)
+    # def serialize_blocks(self, value: Any, _info: Any) -> Any:
+    #     """Serializes the 'blocks' field.
+
+    #     'blocks' is defined as list[ContentBlock] and Pydantic won't serialize
+    #     it automatically since 'ContentBlock' is the base type. We can reuse
+    #     the custom serialization method from additional_kwargs to explicitly
+    #     serialize each block item in the 'blocks' field.
+    #     """
+    #     return self._recursive_serialization(value)
```

Comment:
do we need to keep this commented out code?

Reply:
We absolutely don't! 😅